dao/mongo.go

```diff
@@ -210,3 +227,34 @@ func (m *MongoService) DeletePractitioner(practitionerID string, transactionID s
 
 	return nil, http.StatusNoContent
 }
+
+// AppointPractitioner adds appointment details insolvency case with the specified transactionID and practitionerID
+func (m *MongoService) AppointPractitioner(dao *models.AppointmentResourceDao, transactionID string, practitionerID string) (error, int) {
+
+	collection := m.db.Collection(m.CollectionName)
+
+	// Choose specific practitioner to update
+	filter := bson.M{"transaction_id": transactionID, "data.practitioners.id": practitionerID}
+
+	updateDocument := bson.M{"$set": bson.M{"data.practitioners.$.appointment": dao}}
+
+	update, err := collection.UpdateOne(context.Background(), filter, updateDocument)
+	if err != nil {
+		log.Error(err)
```

Comment:
I would set the error here and then use that created variable to log the error below and then return the error on line 244. Similar to what has been done on line 248 and 254